ods_ci/tasks/Tasks/provision_self_managed_cluster.robot

```diff
@@ -28,11 +28,9 @@ Provision Self-Managed Cluster
     Provision Cluster
     Claim Cluster
     Wait For Cluster To Be Ready
-    Verify Cluster Claim
     Save Cluster Credentials
     Login To Cluster
     Set Cluster Storage
-    [Teardown]  Clean Failed Cluster
```

Comment:
I understand the reasoning behind removing the deprovision step in the case of a legitimate failure (e.g. using the name of a cluster that already exists), but what about the cases in which the provision fails for some other reason? Won't this leave resources pending?

Reply:
I see you have a deprovision step in https://github.com/red-hat-data-services/ods-ci/pull/1078/files#diff-069081f68a5b54944843728104321c2a7f2d596f186734b832ddf7d8265185b7R156, but I'm wondering if that's the only place in which the provisioning can fail

Reply:
yes, since we want to automatically trigger deprovision in case Provision has initially failed, but only if it is due to OCP Install error (i.e. Fatal error).
To further justify the deprovision, this patch also watch Hive Cluster Deployment object, where we can see that the deployed cluster has successfully received the endpoints (API and web-console).